lib/sles4sap/console_redirection.pm

```diff
@@ -0,0 +1,170 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+package sles4sap::console_redirection;
+use strict;
+use warnings;
+use testapi;
+use Carp qw(croak);
+use Exporter qw(import);
+use serial_terminal qw(select_serial_terminal);
+use Regexp::Common qw(net);
+
+=head1 SYNOPSIS
+
+Library that enables console redirection and file transfers from worker based VM to another host.
+Can be used for cases where worker VM is not the target host for API calls and command execution, but serves only as a jumphost.
+
+=cut
+
+our @EXPORT = qw(
+  connect_target_to_serial
+  disconnect_target_from_serial
+  redirection_init
+  check_serial_redirection
+);
+
+my $ssh_opt = '-o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ServerAliveCountMax=120';
+
+=head2 handle_login_prompt
+
+    handle_login_prompt();
+
+B<ssh_user>: Login user
+
+Detects if login prompt appears and types the password.
+In case of ssh keys being in place and command prompt appears, the fucntion does not type anything.
+
+=cut
+
+sub handle_login_prompt {
+    my $pwd = get_var('_SECRET_SUT_PASSWORD', $testapi::password);
+    set_serial_term_prompt();
+    # look for either password prompt or command prompt to appear
+    my $serial_response = wait_serial(qr/Password:\s*$|:~/i, timeout => 20);
```

Comment:
Wasn't intentional, but now I am thinking we could keep it there in case there is change of prompt? Maybe between ssh versions or something like that. wdyt?